packages/devtools_app/lib/src/shared/primitives/utils.dart

```diff
@@ -128,22 +128,67 @@ String printGB(num bytes, {int fractionDigits = 1, bool includeUnit = false}) {
   return output;
 }
 
-/// Converts a [Duration] into a readable text representation in milliseconds.
+enum DurationDisplayUnit {
+  micros('μs'),
+  milliseconds('ms'),
+  seconds('s');
+
+  const DurationDisplayUnit(this.display);
+
+  final String display;
+
+  static DurationDisplayUnit unitFor(int micros) {
+    if (micros < 1000) {
+      return DurationDisplayUnit.micros;
+    } else if (micros < 1000000) {
+      return DurationDisplayUnit.milliseconds;
+    }
+    return DurationDisplayUnit.seconds;
+  }
+}
+
+/// Converts a [Duration] into a readble text representation in the specified
+/// [unit].
 ///
 /// [includeUnit] - whether to include 'ms' at the end of the returned value
 /// [fractionDigits] - how many fraction digits should appear after the decimal
 /// [allowRoundingToZero] - when true, this method may return zero for a very
 /// small number (e.g. '0.0 ms'). When false, this method will return a minimum
 /// value with the less than operator for very small values (e.g. '< 0.1 ms').
 /// The value returned will always respect the specified [fractionDigits].
-String msText(
+String durationText(
   Duration dur, {
+  DurationDisplayUnit? unit,
   bool includeUnit = true,
   int fractionDigits = 1,
   bool allowRoundingToZero = true,
 }) {
-  var durationStr = (dur.inMicroseconds / 1000).toStringAsFixed(fractionDigits);
+  if (!allowRoundingToZero && unit == null) {
+    throw AssertionError('To disable rounding to zero, please specify a unit.');
+  }
 
+  final micros = dur.inMicroseconds;
+  unit ??= DurationDisplayUnit.unitFor(micros);
+  double durationAsDouble;
+  switch (unit) {
+    case DurationDisplayUnit.micros:
+      durationAsDouble = micros.toDouble();
+      break;
+    case DurationDisplayUnit.milliseconds:
+      durationAsDouble = micros / 1000;
+      break;
+    case DurationDisplayUnit.seconds:
+      durationAsDouble = micros / 1000000;
+      break;
+  }
+
+  // Hide any fraction digits when the unit is microseconds, since the
+  // duration displayed will always be a whole number in this case.
+  if (unit == DurationDisplayUnit.micros) {
+    fractionDigits = 0;
+  }
+
```

Comment:
For a duration in microseconds, the fraction digit will always be 0 since the `Duration` class cannot be made more granular than microseconds (i.e. it is impossible to have 0.5 microseconds from a `Duration`).

By default, I think the choice of having `fractionDigits: 1` is the right choice, which was the behavior of the `msText` helper. Defaulting `fractionDigits` to null would change this behavior, which is not what we want IMO.

Reply:
Then we need to document that parameter is not respected in case of microseconds.